### PR TITLE
✨ [amp-story-player] Initial code for reacting to custom UI API from the system layer

### PR DESCRIPTION
**Pull request by Enriqe**
3/n partial for #30031

Reads configuration from the store service and builds the buttons into the system layer according to the publisher-defined configuration.

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story-system-layer.css [hunk 1]
```diff
@@ -299,6 +310,22 @@
...
+  background-image: url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" width="32px" height="32px" viewBox="0 0 32 32" fill="#FFFFFF"><path d="M25.3332 8.54663L23.4532 6.66663L15.9998 14.12L8.5465 6.66663L6.6665 8.54663L14.1198 16L6.6665 23.4533L8.5465 25.3333L15.9998 17.88L23.4532 25.3333L25.3332 23.4533L17.8798 16L25.3332 8.54663Z"/></svg>') !important;
+}
+
+.i-amphtml-story-skip-next {
+  background-image: url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" width="32px" height="32px" viewBox="0 0 32 32" fill="#FFFFFF"><path d="M19.3333 16L8 24V8L19.3333 16ZM24 24H21.3333V8H24V24Z"/></svg>') !important;
+}
+
+.i-amphtml-story-skip-next[disabled] {
+  opacity: 0.3 !important;
+}
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-system-layer.css:327]: Can you factorize to make it work with any `[disabled]` button?

https://github.com/ampproject/amphtml/blob/094908fb576384a602c31ac96411616bb1a8c73a/extensions/amp-story/1.0/amp-story-system-layer.css#L244-L247

**Review by gmajoulet — COMMENTED** [review]
This looks good but will need naming updates from the previous 2 PRs.

File: extensions/amp-story/1.0/amp-story-system-layer.css (hunk 1 above)
Comment (Enriqe) [reply extensions/amp-story/1.0/amp-story-system-layer.css:329]: Done.

**Review by Enriqe — COMMENTED** [review]
I updated this PR with the recent changes to the message name / API. PTAL @gmajoulet

File: extensions/amp-story/1.0/_locales/en.js [hunk 2]
```diff
@@ -75,6 +75,10 @@ const strings = {
       'Label for a button that allows the user to change their ' +
       'choice to consent to providing their cookie access.',
   },
+  [LocalizedStringId.AMP_STORY_CLOSE_BUTTON_LABEL]: {
+    string: 'Close',
+    description: 'Label for a button that closes the story after they click it',
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/_locales/en.js:80]: Nit: translators might not know what a story is. It's worth adding context here, communicating the fact that users open a full page experience and that clicking here will close it and go back to where they were originally

File: extensions/amp-story/1.0/_locales/en.js [hunk 3]
```diff
@@ -310,6 +314,10 @@ const strings = {
       'Label for the sidebar button that pulls up a menu ' +
       'of options for interacting with the story',
   },
+  [LocalizedStringId.AMP_STORY_SKIP_NEXT_BUTTON_LABEL]: {
+    string: 'Skip next',
+    description: 'Label for a button that advances the story to the next page',
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/_locales/en.js:319]: Nit: worth adding context saying that it'll skip the current content to advance to the next element of a carousel. The story / page vocabulary is super confusing to people unfamiliar with AMP Stories

File: extensions/amp-story/1.0/amp-story-system-layer.js [hunk 4]
```diff
@@ -855,6 +913,64 @@ export class SystemLayer {
...
+
+        this.vsync_.mutate(() => {
+          element.parentElement.removeChild(element);
+          startButtonContainer.appendChild(element);
+        });
+      }
+
+      if (control.icon) {
+        setImportantStyles(dev().assertElement(element), {
+          'background-image': `url(${control.icon})`,
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-system-layer.js:968]: Let's clarify a bit more what the `icon` is. Do we want it to always be a URL, or could it be a SVG inlined? Would be cool to have both I assume, how do we allow that?

**Review by gmajoulet — APPROVED** [review]

**Review by jridgewell — APPROVED** [review]

**Pull request merged**